sdk/identity/Azure.Identity/tests/EnvironmentCredentialProviderTests.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Azure.Core;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using Xunit;
+
+namespace Azure.Identity.Tests
+{
+    public static class TestAccessibilityExtensions
+    {
+        public static TokenCredential _credential(this EnvironmentCredentialProvider provider)
+        {
+            return (TokenCredential)typeof(EnvironmentCredentialProvider).GetField("_credential", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(provider);
+        }
+    }
+
+    [CollectionDefinition("AltEnvTests", DisableParallelization = true)]
```

Comment:
You can't put CollectionDefinition on the test class itself AFAIK

Reply:
It looked to me as this was the only place that this can be declared in Xunit

~~~ c#
    //
    // Summary:
    //     Used to declare a test collection container class. The container class gives
    //     developers a place to attach interfaces like Xunit.IClassFixture`1 and Xunit.ICollectionFixture`1
    //     that will be applied to all tests classes that are members of the test collection.
    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = false)]
    public sealed class CollectionDefinitionAttribute : Attribute
~~~

Reply:
It has to be placed on a separate collection class and collection itself referenced by name from the test class. Kind of like this:

``` C#
[CollectionDefinition("EnvironmentTests", DisableParallelization = true)]
public class EnvironmentTestsCollection
{
}

[Collection("EnvironmentTests")]
public class EnvironmentCredentialProviderTests
{
}

```